File: src/com/quotegraph/interfaces/IVector2D.java
package com.quotegraph.interfaces;

/**
 * Interface for 2D double vectors.
 *
 * @author Daniel Peters
 * @version 1.0
 */
public interface IVector2D {
  double getX();

  double getY();

  void add(IVector2D vector);

  void sub(IVector2D vector);

  void mult(double scalar);

  void div(double scalar);

  double mag();

  void normalize();

  void limit(double max);

  void set(double x, double y);

  void set(IVector2D vector);
}

File: src/main/kotlin/ch/peters/daniel/quotegraph/interfaces/IVector2D.kt
package ch.peters.daniel.quotegraph.interfaces

/**
 * Interface for 2D double vectors.
 *
 * @author Daniel Peters
 * @version 1.0
 */
interface IVector2D {
  var x: Double
  var y: Double

  fun add( vector: IVector2D)

  fun sub( vector: IVector2D)

  fun mult(scalar: Double)

  fun div(scalar: Double)

  fun mag(): Double

  fun normalize()

  fun limit(max: Double)

  fun set(valX: Double, valY: Double)

  fun set(vector: IVector2D)
}
